store/sqlstore/job_store.go

```diff
@@ -86,20 +89,21 @@ func (jss SqlJobStore) UpdateStatus(id string, status string) (*model.Job, *mode
 }
 
 func (jss SqlJobStore) UpdateStatusOptimistically(id string, currentStatus string, newStatus string) (bool, *model.AppError) {
-	var startAtClause string
+	sql := jss.getQueryBuilder().
+		Update("Jobs").
+		Set("LastActivityAt", model.GetMillis()).
+		Set("Status", newStatus).
+		Where(sq.Eq{"Id": id, "Status": currentStatus})
+
 	if newStatus == model.JOB_STATUS_IN_PROGRESS {
-		startAtClause = "StartAt = :StartAt,"
+		sql = sql.Set("StartAt", model.GetMillis())
 	}
-	query := "UPDATE Jobs SET " + startAtClause + " Status = :NewStatus, LastActivityAt = :LastActivityAt WHERE Id = :Id AND Status = :OldStatus"
-	params := map[string]interface{}{
-		"Id":             id,
-		"OldStatus":      currentStatus,
-		"NewStatus":      newStatus,
-		"StartAt":        model.GetMillis(),
-		"LastActivityAt": model.GetMillis(),
+	query, args, err := sql.ToSql()
+	if err != nil {
+		return false, model.NewAppError("SqlJobStore.UpdateStatus", "store.sql.build_query.app_error", nil, err.Error(), http.StatusInternalServerError)
 	}
 
-	sqlResult, err := jss.GetMaster().Exec(query, params)
+	sqlResult, err := jss.GetMaster().Exec(query, args...)
 	if err != nil {
 		return false, model.NewAppError("SqlJobStore.UpdateStatus", "store.sql_job.update.app_error", nil, "id="+id+", "+err.Error(), http.StatusInternalServerError)
```

Comment:
Although is not part of this PR, for consistency this one should be:

```suggestion
		return false, model.NewAppError("SqlJobStore.UpdateStatusOptimistically", "store.sql_job.update.app_error", nil, "id="+id+", "+err.Error(), http.StatusInternalServerError)
```